### PR TITLE
ref(searchBar): Add support for different input sizes

### PR DESCRIPTION
**Pull request by vuluongj20**
Update `SearchBar` to automatically adjust its own padding and the search icon's size based on the current input size (modifiable via the `size` prop).

**`size="md"`**
Screen Shot 2022-08-17 at 2 28 47 PM

**`size="sm"`**
Screen Shot 2022-08-17 at 2 29 01 PM

**`size="xs"`**
Screen Shot 2022-08-17 at 2 29 10 PM

### CODE REVIEW

File: static/app/components/profiling/flamegraphSearch.tsx [hunk 1]
```diff
@@ -312,6 +312,7 @@ function FlamegraphSearch({
 
   return (
     <StyledSearchBar
+      size="xs"
```
Comment (vuluongj20) [inline static/app/components/profiling/flamegraphSearch.tsx:315]: **Before:**
Screen Shot 2022-08-17 at 11 47 40 AM

**After:**
Screen Shot 2022-08-17 at 11 45 59 AM

File: static/app/components/events/interfaces/searchBarAction.tsx [hunk 2]
```diff
@@ -88,35 +88,20 @@ const Wrapper = styled('div')`
...
       }
     `}
 
   @media (max-width: ${p => p.theme.breakpoints.small}) {
-    .search-input,
-    .search-input:focus {
+    input {
       border-radius: ${p => p.theme.borderRadius};
       border-left-width: 1px;
     }
```
Comment (vuluongj20) [inline static/app/components/events/interfaces/searchBarAction.tsx:107]: Same styles, less code ✨

image

File: static/app/components/searchBar.tsx [hunk 3]
```diff
@@ -10,151 +10,118 @@ import {t} from 'sentry/locale';
...
-
-  onQueryFocus = () => {
-    this.setState({
-      dropdownVisible: true,
-    });
-  };
-
-  onQueryBlur = () => {
-    this.setState({dropdownVisible: false});
-  };
```
Comment (vuluongj20) [inline static/app/components/searchBar.tsx:74]: `dropdownVisible` is a legacy state that does nothing

File: static/app/components/searchBar.tsx [hunk 4]
```diff
@@ -10,151 +10,118 @@ import {t} from 'sentry/locale';
...
-      query: _q,
-      defaultQuery,
-      onChange: _oC,
-      onSearch: _oS,
-      ...inputProps
-    } = this.props;
+  function onSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    inputRef.current?.blur();
+    callIfFunction(onSearch, query);
```
Comment (evanpurkhiser) [inline static/app/components/searchBar.tsx:43]:
```suggestion
   onSearch?.(query);
```

File: static/app/components/searchBar.tsx [hunk 5]
```diff
@@ -10,151 +10,118 @@ import {t} from 'sentry/locale';
...
-            )}
-          </div>
-        </form>
-      </div>
-    );
+  function clearSearch() {
+    setQuery('');
+    callIfFunction(onChange, '');
+    callIfFunction(onSearch, '');
   }
```
Comment (evanpurkhiser) [inline static/app/components/searchBar.tsx:50]:
```suggestion
    onChange?.('');
    onSearch?.('');
  }
```

File: static/app/components/searchBar.tsx [hunk 6]
```diff
@@ -10,151 +10,118 @@ import {t} from 'sentry/locale';
...
-                size="xs"
-                icon={<IconClose />}
-                aria-label={t('Clear')}
-              />
-            )}
-          </div>
-        </form>
-      </div>
-    );
+  function clearSearch() {
```
Comment (evanpurkhiser) [inline static/app/components/searchBar.tsx:46]: probably want to `useCallback` for this

File: static/app/components/searchBar.tsx [hunk 7]
```diff
@@ -10,151 +10,118 @@ import {t} from 'sentry/locale';
...
+    setQuery('');
+    callIfFunction(onChange, '');
+    callIfFunction(onSearch, '');
   }
+
+  const iconSize = useMemo(() => (size === 'xs' ? 'xs' : 'sm'), [size]);
```
Comment (evanpurkhiser) [inline static/app/components/searchBar.tsx:52]: do you actually need to memoize here?

Comment (evanpurkhiser) [reply static/app/components/searchBar.tsx:52]: this isn't really expensive to compute, or something we need to worry about it's identity changing.

File: static/app/components/searchBar.tsx (hunk 6 above)
Comment (vuluongj20) [reply static/app/components/searchBar.tsx]: gotcha

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**